plugin/src/main/java/dev/regadas/trino/pubsub/listener/PubSubEventListener.java

```diff
@@ -28,88 +21,76 @@ public final class PubSubEventListener implements EventListener, AutoCloseable {
-                Publisher.newBuilder(config.topicName())
-                        .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
-                        .setEnableCompression(true)
-                        .build();
-
-        var encoder = MessageEncoder.create(config.encoding());
-        return new PubSubEventListener(config, publisher, encoder);
+                PubSubPublisher.create(
+                        config.projectId(),
+                        config.topicId(),
+                        config.encoding(),
+                        config.credentialsFilePath());
+        return new PubSubEventListener(config, publisher);
     }
 
     @Override
     public void queryCreated(QueryCreatedEvent event) {
         if (config.trackQueryCreatedEvent()) {
-            publish(SchemaHelpers.from(event));
+            publish(SchemaHelpers.from(event), pubSubInfo.queryCreated());
         }
     }
 
     @Override
     public void queryCompleted(QueryCompletedEvent event) {
         if (config.trackQueryCompletedEvent()) {
-            publish(SchemaHelpers.from(event));
+            publish(SchemaHelpers.from(event), pubSubInfo.queryCompleted());
         }
     }
 
     @Override
     public void splitCompleted(SplitCompletedEvent event) {
         if (config.trackSplitCompletedEvent()) {
-            publish(SchemaHelpers.from(event));
+            publish(SchemaHelpers.from(event), pubSubInfo.splitCompleted());
         }
     }
 
-    void publish(Message event) {
+    void publish(Message event, PubSubCounters counters) {
         try {
-            var data = encoder.encode(event);
-            var message = PubsubMessage.newBuilder().setData(ByteString.copyFrom(data)).build();
+            counters.attempts().incrementAndGet();
+            var future = publisher.publish(event);
 
-            var future = publisher.publish(message);
-
-            ApiFutures.addCallback(
-                    future,
-                    new ApiFutureCallback<>() {
-                        public void onSuccess(String id) {
+            future.whenComplete(
+                    (id, t) -> {
+                        if (t == null) {
+                            counters.successful().incrementAndGet();
                             LOG.log(Level.ALL, "published event with id: " + id);
-                        }
-
-                        public void onFailure(Throwable t) {
+                        } else {
+                            counters.failure().incrementAndGet();
                             LOG.log(Level.SEVERE, "Failed to publish event", t);
                         }
-                    },
-                    MoreExecutors.directExecutor());
-
+                    });
         } catch (Exception e) {
+            counters.failure().incrementAndGet();
             LOG.log(Level.SEVERE, "Failed to publish", e);
         }
     }
 
     @Override
     public void close() {
-        if (publisher != null) {
```

Comment:
publisher is never null, we do  `this.publisher = requireNonNull(publisher, "publisher is null");`